source/fundamentals/crud/bulk.txt

```diff
@@ -0,0 +1,315 @@
+
+Replace Operation
+~~~~~~~~~~~~~~~~~
+
+To perform a replace operation, create an ``ReplaceOneModel`` specifying
+the document you want to replace with the replacement document. 
+
+.. note::
+
+   When performing a ``bulkWrite()``, the ``ReplaceOneModel`` cannot
+   make changes to a document that violate unique index constraints on
+   the collection. Also, the model does not replace a document if there
+   are no matches to your query filter. 
+
+Example
+```````
+
+The following example creates a ``ReplaceOneModel`` to
+replace a document where the ``_id`` is "1" to a document with an
+additional field ``x`` with the value "4": 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/BulkWrite.java
+   :language: java
+   :dedent:
+   :start-after: begin replaceDocumentsExample
+   :end-before: end replaceDocumentsExample
+
+For additional information on the classes and methods mentioned in this
+section, see the following resources: 
+
+- :java-core-api:`ReplaceOneModel <com/mongodb/client/model/ReplaceOneModel.html>` API Documentation
+- :manual:`Unique indexes </core/index-unique/>` Server Manual Explanation
+
+Update Operation
+~~~~~~~~~~~~~~~~
+
+To perform an update operation, create an ``UpdateOneModel`` or an
+``UpdateManyModel`` specifying the document you want to update with what
+the updates are.
+
+The ``UpdateOneModel`` updates one document that matches your query
+filter and the ``UpdateManyModel`` updates all the documents that
+match your query filter. 
+
+.. note::
+
+   When performing a ``bulkWrite()``, the ``UpdateOneModel`` and
+   ``UpdateManyModel`` cannot make changes to a document that violate
+   unique index constraints on the collection. Also, the models do not
+   update any documents if there are no matches to your query filter. 
+
+Example
+```````
+
+The following example creates an ``UpdateOneModel`` to update
+a document where the ``_id`` is "2" to a document with an additional
+field ``x`` with the value "8": 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/BulkWrite.java
+   :language: java
+   :dedent:
+   :start-after: begin updateDocumentsExample
+   :end-before: end updateDocumentsExample
+
+For additional information on the classes and methods mentioned in this
+section, see the following resources: 
+
+- :java-core-api:`UpdateOneModel <com/mongodb/client/model/UpdateOneModel.html>` API Documentation
+- :java-core-api:`UpdateManyModel <com/mongodb/client/model/UpdateManyModel.html>` API Documentation
+- :manual:`unique indexes </core/index-unique/>` Server Manual Explanation
+
+Delete Operation
+~~~~~~~~~~~~~~~~
+
+To perform a delete operation, create a ``DeleteOneModel`` or a
+``DeleteManyModel`` specifying the document you want to delete. 
+
+The ``DeleteOneModel`` deletes one document that matches your query
+filter and the ``DeleteManyModel`` deletes all the documents that
+match your query filter. 
+
```

Comment:
same as before, is it more accurate to state it will delete the `first` document it matches?